fix(core): fix managed field issue + tests

d-sooter opened the pull request:
Fix issue with managed fields and added test for inserts on managed fields

vobu commented:
errrmmm, sorry to grätsch in here, but where exactly is the deep insert test?
can only see 
- src code reformatting (nooo..., prettier!!!)
- fixes to existing tests
am i missing something?

d-sooter commented:
I fixed an issue with the managed fields. added them to the data model and checked that they were being filled in the tests. That fixed the deep insert issue. But I will add a test case for it at well so we have that covered too.

vobu commented:
excellent, some `request.post(...).send{ /* deeply nested json obj */ })` will certainly help assure quality.
you'll certainly also have to change `schema.cds` of the sample beershop, most prominently replace `association to` with `composition of` - otherwise deeps inserts won't work anyway.
